pkg/action/cl001/ac008/executor.go

```diff
@@ -2,8 +2,279 @@ package ac008
+	kiamAgentLabelSelector  = "app=kiam,component=kiam-agent"
+
+	masterNodeLabelSelector = "kubernetes.io/role=master"
+	workerNodeLabelSelector = "kubernetes.io/role=worker"
+
+	checkJobRetryLimit = 15
+	checkJobMaxWait    = time.Second * 10
+
+	draughtsmanNamespace                  = "draughtsman"
+	draughtsmanConfigMapName              = "draughtsman-values-configmap"
+	draughtsmanConfigMapDataKey           = "values"
+	draughtsmanConfigMapDockerRegistryKey = "Installation.V1.Registry.Domain"
 )
 
 func (e *Executor) execute(ctx context.Context) error {
+	var err error
+
+	var cpClients k8sclient.Interface
+	{
+		c := client.ControlPlaneConfig{
+			Logger: e.logger,
+
+			KubeConfig: env.KubeConfig(),
+		}
+
+		cpClients, err = client.NewControlPlane(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var tcClients k8sclient.Interface
+	{
+		c := client.TenantClusterConfig{
+			ControlPlane: cpClients,
+			Logger:       e.logger,
+
+			Scope: e.scope,
+		}
+
+		tcClients, err = client.NewTenantCluster(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	// awsRegion is necessary to execute aws-cli call
+	var awsRegion string
+	{
+		var cr infrastructurev1alpha2.AWSCluster
+		{
+			var list infrastructurev1alpha2.AWSClusterList
+			err := cpClients.CtrlClient().List(
+				ctx,
+				&list,
+				k8sruntimeclient.InNamespace(cr.GetNamespace()),
+				k8sruntimeclient.MatchingLabels{label.Cluster: e.tenantCluster},
+			)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			if len(list.Items) == 0 {
+				return microerror.Mask(notFoundError)
+			}
+			if len(list.Items) > 1 {
+				return microerror.Mask(tooManyCRsError)
+			}
+
+			cr = list.Items[0]
+		}
+
+		awsRegion = cr.Spec.Provider.Region
+	}
+
+	// dockerRegistry is needed in order to spawn pod with proper docker image that will execute aws-cli call
+	var dockerRegistry string
+	{
+		dockerRegistry, err = e.fetchDockerRegistry(ctx, cpClients.K8sClient())
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	err = e.checkTLSCerts(ctx, tcClients.K8sClient())
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = e.checkKiamPods(ctx, tcClients.K8sClient())
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = e.testAWSApiCalls(ctx, tcClients.K8sClient(), awsRegion, dockerRegistry)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
 	return nil
 }
+
+// checkTLSCerts Ensures that kiam  tls certs are created.
+var kiamTlSCertSecretNames = []string{"kiam-agent-tls", "kiam-ca-tls", "kiam-server-tls"}
+
+func (e *Executor) checkTLSCerts(ctx context.Context, tcClient kubernetes.Interface) error {
+	for _, secret := range kiamTlSCertSecretNames {
+		_, err := tcClient.CoreV1().Secrets(kubeSystemNamespace).Get(ctx, secret, metav1.GetOptions{})
+
+		if err != nil {
+			e.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("Could not find secret %s in namespace %s", secret, kubeSystemNamespace))
+			return microerror.Mask(err)
```

Comment:
Is it good enough to do only error handling? The error should be displayed properly. Additional logging feels wrong to me. When the error does not transport the right information we can use our own error and annotate it with the necessary information.

Reply:
removed